grpc-sys/build.rs

```diff
@@ -172,7 +172,11 @@ fn build_grpc(cc: &mut cc::Build, library: &str) {
         config.define("gRPC_BUILD_CODEGEN", "false");
         // We don't need to build benchmarks.
         config.define("gRPC_BENCHMARK_PROVIDER", "none");
-        config.define("gRPC_SSL_PROVIDER", "package");
+
+        if cfg!(feature = "secure") {
+            config.define("gRPC_SSL_PROVIDER", "package");
+        }
```

Comment:
why not just define the cmake config regardless of the `secure` feature

Reply:
@hug-dev can you add a comment here?

Reply:
I did that to fix a failing test on Mac OS, after [your recommendation](https://github.com/tikv/grpc-rs/pull/537#issuecomment-866605705).
Would the following comment be ok?

```
// OpenSSL is needed for gRPC compilation when the secure feature is on.
```

Reply:
How about
```
// `package` should only be set for secure feature, otherwise cmake will always search for ssl library.
```

Reply:
Done!